client/src/components/testUtilities.tsx

```diff
@@ -89,7 +93,12 @@ export const withMockFetch = async (
       expectedRequest.url === url &&
       (expectedRequest.skipBody || equal(expectedRequest.options, options))
     ) {
-      return new Response(JSON.stringify(expectedRequest.response))
+      return expectedRequest.error
+        ? new Response(
+            JSON.stringify(expectedRequest.response),
+            expectedRequest.error
+          )
+        : new Response(JSON.stringify(expectedRequest.response))
```

Comment:
nice extension, seems very useful

Reply:
@jonahkagan Yeah, we can test error responses now, which is super handy.

I ran into an issue with comparing the formData objects because they'd never match the mock (literally impossible, since the code creates and sends a new one each run), so I was wondering if passing in a matcher function that defaults to the current behavior could resolve that, but I bypassed it for now with just a skip flag.

Reply:
Hmm what's an example test where that's happening? I'm curious to check it out

Reply:
Here's where I needed to use it: https://github.com/votingworks/arlo/blob/main/client/src/components/MultiJurisdictionAudit/index.test.tsx#L373

Basically, the file upload process creates a new formData object when it's submitting the form, and while I could create another formData object using the same file in the test, it's two different objects and they won't equal each other.

Reply:
Is it possible to detect when the `body` is a `FormData` object and then either skip it or, better, pull out the actual data as a string and check it?

Reply:
@jonahkagan Would you advocate doing that inside your test function? Or passing a function into the test function that itself handles that kind of conditional management?

Reply:
It makes sense to me to have it be part of `withMockFetch`, since form data is a common kind of request body

Reply:
@jonahkagan Fair enough. We only ever use the `FormData` object when we're uploading a file, though, which we only do in a couple places. I think if it's possible to extract the data of both and compare those, though, it'd be valuable to check for that and do it automatically.

Reply:
@jonahkagan So we can detect that it's a `FormData` object, but extracting the data is an asynchronous process, and I think that's why it's not working. I've tried a bunch of different ways, but I think it's something we'll have to return to later, or if one of us gets a stroke of inspiration. We can use the `skipBody` option for now without concern, I think, since comparing the content of the file is not actually testing any of the functionality of the app.